### ISSUE TITLE
Dotdigitalgroup_Email data install fails on install/upgrade for split database installs 

### ISSUE THREAD

**Issue by navarr** [dotmailer/dotmailer-magento2-extension#559]
### Preconditions (*)
1. Magento 2.1.5 installed

### Steps to reproduce (*)
1. Split sales and quote databases
2. Upgrade to Magento 2.3.4

### Expected result (*)
1. Upgrade goes smoothly with no errors

### Actual result (*)
```
Module 'Dotdigitalgroup_Email':
Installing data... SQLSTATE[42S02]: Base table or view not found: 1146 Table 'magento.sales_order' doesn't exist, query was: INSERT INTO `email_order` (`order_id`, `quote_id`, `store_id`, `created_at`, `updated_at`, `order_status`) SELECT `sales_order`.`entity_id` AS `order_id`, `sales_order`.`quote_id`, `sales_order`.`store_id`, `sales_order`.`created_at`, `sales_order`.`updated_at`, `sales_order`.`status` AS `order_status` FROM `sales_order` ORDER BY `order_id` ASC
 LIMIT 500
```

**Comment by sta1r** [comment]
Hi @navarr thanks for raising this. We've replicated the fault and expect to release a fix with our 4.5.2 release on 22 April.

**Comment by sta1r** [comment]
@navarr Hi again, we have released a fix in our v4.5.2: https://github.com/dotmailer/dotmailer-magento2-extension/releases/tag/v4.5.2
